File: src/marcas-neumaticos/dto/create-marcas-neumatico.dto.ts
import { IsNotEmpty, IsString, MaxLength } from "class-validator";

export class CreateMarcasNeumaticoDto {

    @IsString()
    @IsNotEmpty({ message: 'La marca es requerida' })
    @MaxLength(15, { message: 'La marca no puede sobrepasar los 15 caracteres' })
    marca: string;
}

File: src/main/java/com/tallermecanico/dto/request/CreateMarcasNeumaticoDto.java
package com.tallermecanico.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class CreateMarcasNeumaticoDto {

    @NotBlank(message = "La marca es requerida")
    @Size(max = 15, message = "La marca no puede exceder 15 caracteres")
    private String marca;

    // Constructors
    public CreateMarcasNeumaticoDto() {
    }

    public CreateMarcasNeumaticoDto(String marca) {
        this.marca = marca;
    }

    // Getters and Setters
    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }
}
